cmd/stack/proxy.go

```diff
@@ -63,6 +66,49 @@ func (c *StackProxyController) GetStore() *StackProxyStore {
 	return c.store
 }
 
+// corsMiddleware creates a CORS middleware that handles preflight requests and adds CORS headers
+func corsMiddleware(allowedOrigins []string, next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		origin := r.Header.Get("Origin")
+
+		// Check if origin is allowed
+		originAllowed := false
+		allowWildcard := false
+		for _, allowedOrigin := range allowedOrigins {
+			if allowedOrigin == "*" {
+				originAllowed = true
+				allowWildcard = true
+				break
+			} else if allowedOrigin == origin {
+				originAllowed = true
+				break
+			}
+		}
+
+		// Set CORS headers only for allowed origins
+		if originAllowed {
+			if allowWildcard {
+				w.Header().Set("Access-Control-Allow-Origin", "*")
+			} else if origin != "" {
+				w.Header().Set("Access-Control-Allow-Origin", origin)
+			}
+
+			w.Header().Set("Access-Control-Allow-Methods", "GET, POST, PUT, DELETE, PATCH, OPTIONS")
+			w.Header().Set("Access-Control-Allow-Headers", "Accept, Authorization, Content-Type, X-CSRF-Token, X-Requested-With")
+			w.Header().Set("Access-Control-Allow-Credentials", "true")
+			w.Header().Set("Access-Control-Max-Age", "86400") // 24 hours
+		}
```

Comment:
I think it became more complicated 😅. Better to have fewer temporary variables & fewer if branches, something like this:

```suggestion
		var originStr string
		for _, allowedOrigin := range allowedOrigins {
			if allowedOrigin == "*" {
				originStr = "*"
				break
			} else if allowedOrigin == origin {
				originStr = origin
				break
			}
		}

		// Set CORS headers only for allowed origins
		if originStr != "" {
			w.Header().Set("Access-Control-Allow-Origin", originStr)
			w.Header().Set("Access-Control-Allow-Methods", "GET, POST, PUT, DELETE, PATCH, OPTIONS")
			w.Header().Set("Access-Control-Allow-Headers", "Accept, Authorization, Content-Type, X-CSRF-Token, X-Requested-With")
			w.Header().Set("Access-Control-Allow-Credentials", "true")
			w.Header().Set("Access-Control-Max-Age", "86400") // 24 hours
		}
```